samples/sensor/hx711/src/main.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2020 Kalyan Sriram <kalyan@coderkalyan.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+
+#define ADC_NODE DT_INST(0, avia_hx711)
+
+#if DT_NODE_HAS_STATUS(ADC_NODE, okay)
+#define ADC_LABEL DT_LABEL(ADC_NODE)
+#else
+#error "Unsupported board."
+#endif
+
+static const char *now_str(void)
+{
+	static char buf[16]; /* ...HH:MM:SS.MMM */
+	uint32_t now = k_uptime_get_32();
+	unsigned int ms;
+	unsigned int s;
+	unsigned int min;
+	unsigned int h;
+
+	ms = now % MSEC_PER_SEC;
+	now /= MSEC_PER_SEC;
+	s = now % 60U;
+	now /= 60U;
+	min = now % 60U;
+	now /= 60U;
+	h = now;
+
+	snprintf(buf, sizeof(buf), "%u:%02u:%02u.%03u",
+			h, min, s, ms);
+	return buf;
+}
+
+static int process_hx711(const struct device *dev)
+{
+	struct sensor_value load_calib;
+	struct sensor_value load_raw;
+	int rc = sensor_sample_fetch(dev);
+
+	if (rc == 0) {
+		rc = sensor_channel_get(dev, SENSOR_CHAN_FORCE, &load_calib);
+	}
+	if (rc == 0) {
+		rc = sensor_channel_get(dev, SENSOR_CHAN_PRIV_START, &load_raw);
+	}
+
+	if (rc == 0) {
+		printf("[%s]:\n"
+			   "calibrated: %f grams\n"
```

Comment:
IMHO, printing info on single line looks better:

[0:02:56.911]: calibrated: -91053.000000 grams, raw: -91053.000000 grams
[0:02:58.917]: calibrated: -90883.000000 grams, raw: -90883.000000 grams
[0:03:00.923]: calibrated: -90901.000000 grams, raw: -90901.000000 grams

instead of:

[0:01:32.630]:
calibrated: -91254.000000 grams
raw: -91254.000000 grams
[0:01:34.637]:
calibrated: -91184.000000 grams
raw: -91184.000000 grams